core/drivers/clk/clk_dt.c

```diff
@@ -179,6 +179,7 @@ static void parse_assigned_clock(const void *fdt, int nodeoffset)
 		}
 
 		clock_idx++;
+		clk = NULL;
```

Comment:
This obviously fixes the issue, but I wonder if the root of the problem isn't further up around lines 158-162.
Perhaps something like:
``` C
if (res)
     return;
assert(clk);
```

Reply:
Updated, thank you for your comment.